cmds/report.c

```diff
@@ -20,6 +21,227 @@ enum {
+	.print   = print_total_time_min,
+	.list    = LIST_HEAD_INIT(field_total_time_min.list),
+};
+
+static struct display_field field_total_time_max = {
+	.id      = REPORT_F_TOTAL_TIME_MAX,
+	.name    = "total-time-max",
+	.alias   = "total-max",
+	.header  = " Total max",
+	.length  = 10,
+	.print   = print_total_time_max,
+	.list    = LIST_HEAD_INIT(field_total_time_max.list),
+};
+
+static struct display_field field_self_time = {
+	.id      = REPORT_F_SELF_TIME,
+	.name    = "self-time",
+	.alias   = "self",
+	.header  = " Self time",
+	.length  = 10,
+	.print   = print_self_time,
+	.list    = LIST_HEAD_INIT(field_self_time.list),
+};
+
+static struct display_field field_self_time_avg = {
+	.id      = REPORT_F_SELF_TIME_AVG,
+	.name    = "self-time-avg",
+	.alias   = "self-avg",
+	.header  = "  Self avg",
+	.length  = 10,
+	.print   = print_self_time_avg,
+	.list    = LIST_HEAD_INIT(field_self_time_avg.list),
+};
+
+static struct display_field field_self_time_min = {
+	.id      = REPORT_F_SELF_TIME_MIN,
+	.name    = "self-time-min",
+	.alias   = "self-min",
+	.header  = "  Self min",
+	.length  = 10,
+	.print   = print_self_time_min,
+	.list    = LIST_HEAD_INIT(field_self_time_min.list),
+};
+
+static struct display_field field_self_time_max = {
+	.id      = REPORT_F_SELF_TIME_MAX,
+	.name    = "self-time-max",
+	.alias   = "self-max",
+	.header  = "  Self max",
+	.length  = 10,
+	.print   = print_self_time_max,
+	.list    = LIST_HEAD_INIT(field_self_time_max.list),
+};
+
+static struct display_field field_call = {
+	.id      = REPORT_F_CALL,
+	.name    = "call",
+	.alias   = "call",
+	.header  = "     Calls",
+	.length  = 10,
+	.print   = print_call,
+	.list    = LIST_HEAD_INIT(field_call.list),
+};
+
+/* index of this table should be matched to display_field_id */
+static struct display_field *field_table[] = {
+	&field_total_time,
+	&field_total_time_avg,
+	&field_total_time_min,
+	&field_total_time_max,
+	&field_self_time,
+	&field_self_time_avg,
+	&field_self_time_min,
+	&field_self_time_max,
+	&field_call,
+};
+
+static void setup_default_field(struct list_head *fields, struct opts *opts)
+{
+	add_field(fields, field_table[REPORT_F_TOTAL_TIME]);
+	add_field(fields, field_table[REPORT_F_SELF_TIME]);
+	add_field(fields, field_table[REPORT_F_CALL]);
+}
+
+static void setup_avg_total_field(struct list_head *fields, struct opts *opts)
+{
+	add_field(fields, field_table[REPORT_F_TOTAL_TIME_AVG]);
+	add_field(fields, field_table[REPORT_F_TOTAL_TIME_MIN]);
+	add_field(fields, field_table[REPORT_F_TOTAL_TIME_MAX]);
+}
+
+static void setup_avg_self_field(struct list_head *fields, struct opts *opts)
+{
+	add_field(fields, field_table[REPORT_F_SELF_TIME_AVG]);
+	add_field(fields, field_table[REPORT_F_SELF_TIME_MIN]);
+	add_field(fields, field_table[REPORT_F_SELF_TIME_MAX]);
+}
+
+static void print_field(struct uftrace_report_node *node)
+{
+	struct field_data fd = {
+		.arg = node,
+	};
+
+	print_field_data(&output_fields, &fd, 2);
+}
```

Comment:
Please move all these code to utils/report.c file like sorting code.  It might be possible we share some code with `tui` command.

Reply:
ok.